plugin.rb

```diff
@@ -11,199 +11,49 @@
-    end
-
-    def create
-      guardian.ensure_can_edit_canned_replies!
-
-      title   = params.require(:title)
-      content = params.require(:content)
-      user_id = current_user.id
-
-      record = CannedReply::Reply.add(user_id, title, content)
-      render json: record
-    end
-
-    def destroy
-      guardian.ensure_can_edit_canned_replies!
-
-      reply_id = params.require(:id)
-      user_id  = current_user.id
-      record = CannedReply::Reply.remove(user_id, reply_id)
-
-      render json: record
-    end
-
-    def reply
-      reply_id = params.require(:id)
-      user_id  = current_user.id
-
-      record = CannedReply::Reply.get_reply(user_id, reply_id)
-      render json: record
-    end
-
-    def update
-      guardian.ensure_can_edit_canned_replies!
-
-      reply_id = params.require(:id)
-      title = params.require(:title)
-      content = params.require(:content)
-      user_id = current_user.id
-
-      record = CannedReply::Reply.edit(user_id, reply_id, title, content)
-      render json: record
-    end
-
-    def use
-      reply_id = params.require(:id)
-      user_id  = current_user.id
-      record = CannedReply::Reply.use(user_id, reply_id)
-
-      render json: record
-    end
-
-    def index
-      user_id = current_user.id
-      replies = CannedReply::Reply.all(user_id)
-
-      render json: { replies: replies }
-    end
-  end
-
-  add_to_class(:user, :can_edit_canned_replies?) do
-    return true if staff?
-    return true if SiteSetting.canned_replies_everyone_can_edit
-    group_list = SiteSetting.canned_replies_groups.split("|").map(&:downcase)
-    groups.any? { |group| group_list.include?(group.name.downcase) }
-  end
-
-  add_to_class(:user, :can_use_canned_replies?) do
-    return true if staff?
-    return true if SiteSetting.canned_replies_everyone_enabled
-    group_list = SiteSetting.canned_replies_groups.split("|").map(&:downcase)
-    groups.any? { |group| group_list.include?(group.name.downcase) }
-  end
-
-  add_to_class(:guardian, :can_edit_canned_replies?) do
-    user && user.can_edit_canned_replies?
-  end
-
-  add_to_class(:guardian, :can_use_canned_replies?) do
-    user && user.can_use_canned_replies?
+    User.class_eval { prepend DiscourseCannedReplies::UserExtension }
   end
 
   add_to_serializer(:current_user, :can_use_canned_replies) do
     object.can_use_canned_replies?
   end
 
-  add_to_serializer(:current_user, :can_edit_canned_replies) do
-    object.can_edit_canned_replies?
+  Discourse::Application.routes.append do
+    mount ::DiscourseCannedReplies::Engine, at: '/canned_replies'
   end
 
-  CannedReply::Engine.routes.draw do
-    resources :canned_replies, path: '/', only: [:index, :create, :destroy, :update] do
-      member do
-        get "reply"
-        patch "use"
-      end
+  DiscourseCannedReplies::Engine.routes.draw do
+    resources :canned_replies, path: '/', only: [:index] do
+      member { patch 'use' }
```

Comment:
Just a note: In core we specifically disable PATCH requests (see: https://github.com/discourse/discourse/blob/main/config/routes.rb#L11-L11)

I see it was using `PATCH` previously, but maybe it would be better to change it to POST? 🤔 Just to be consistent with core.

Reply:
Fixed in 23b7c056f160ca136d9389af27d501f2600cd30a